airbyte-ci/connectors/pipelines/pipelines/dagger/actions/gradle.py

```diff
@@ -0,0 +1,230 @@
+        rsync_gradle_home_to_cache_volume = f"(rsync -a --stats {self.GRADLE_HOME_PATH}/ {self.GRADLE_DEP_CACHE_PATH} || true)"
+        # Running a gradle task like "help" with these arguments will trigger updating all dependencies.
+        # When the cache is cold, this downloads many gigabytes of jars and poms from all over the internet.
+        resolve_deps_and_write_checksum = self._get_gradle_command("help", *warm_dependency_cache_args)
+        build_and_publish_local_cdk = self._get_gradle_command(":airbyte-cdk:java:airbyte-cdk:publishSnapshotIfNeeded")
+
+        gradle_commands = [resolve_deps_and_write_checksum]
+        if build_cdk:
+            gradle_commands.append(build_and_publish_local_cdk)
+
+        commands = [
+            rsync_cache_volume_to_gradle_home,
+            *gradle_commands,
+            rsync_gradle_home_to_cache_volume,
+        ]
+
+        def wrapper(gradle_container: dagger.Container) -> dagger.Container:
+            return gradle_container.with_exec(sh_dash_c(commands))
+
+        return wrapper
+
+    def _get_gradle_command(self, task: str, *args) -> str:
+        """Get the gradlew command to run a given task with the given arguments."""
+        return f"./gradlew {' '.join(self.DEFAULT_GRADLE_TASK_OPTIONS + args)} {task}"
+
+    async def __aenter__(self) -> GradleTaskExecutor:
+        """Enter the context manager and return a gradle task executor with a base gradle container with a seeded cache and S3 build cache set up.
+
+        Returns:
+            GradleTaskExecutor: A gradle task executor with a base gradle container.
+        """
+        warm_dependency_cache_args = ["--write-verification-metadata", "sha256", "--dry-run"] if not self.local_execution else ["--dry-run"]
+
+        self.gradle_container = await (
+            self.get_base_gradle_container()
+            .with_(self.set_s3_build_cache_credentials(self.s3_build_cache_access_key_id, self.s3_build_cache_secret_key))
+            .with_mounted_directory(self.workdir, self.gradle_project_dir)
+            .with_(
+                self.warm_up_dependency_cache(self.GRADLE_DEP_CACHE_PATH, self.GRADLE_HOME_PATH, self.build_cdk, warm_dependency_cache_args)
+            )
+        )
+
+        return self
+
+    async def run_task(self, task_name: str, *args) -> dagger.Container:
+        """Run a gradle task with the given arguments.
+
+        Args:
+            task_name (str): The name of the gradle task to run.
+            *args: The arguments to pass to the gradle task.
+
+        Raises:
+            Exception: If gradlew is not found in the current directory.
+
+        Returns:
+            dagger.Container: The gradle container with the task executed.
+        """
+        if "gradlew" not in await self.gradle_container.directory(".").entries():
+            raise Exception("gradlew not found in the current directory, please mount a directory with gradlew to the container.")
```

Comment:
Gratuitous complexity? The failure will become apparent rather quickly, won't it?